### PR TITLE
test: Code Coverage: Add-Edit Expense Page #1

### PR DESCRIPTION
**Pull request by jayfyle**
### <samp>🤖 Generated by Copilot at e0a62de</samp>

This pull request adds mock data files for cost centers and duplicate sets to `src/app/core/mock-data`. The mock data is used for testing the new features of cost center selection and duplicate detection in the mobile app.

### <samp>🤖 Generated by Copilot at e0a62de</samp>

> _Mock data arrays_
> _`cost-centers` and `duplicate-sets`_
> _Autumn of testing_

### <samp>🤖 Generated by Copilot at e0a62de</samp>

*  Add mock data for cost centers and duplicate sets features ([link](https://github.com/fylein/fyle-mobile-app/pull/2123/files?diff=unified&w=0#diff-7de0329a6c26862e485c2274615a2f69250e4144bdf3abd27cc61d704384ec91R177-R205), [link](https://github.com/fylein/fyle-mobile-app/pull/2123/files?diff=unified&w=0#diff-08994ab67f55c1a7ea9ee90f7380a90fbd85abd94db9dbeee14f217f2cb1ebdfR1-R6))
  - Create `src/app/core/mock-data/cost-centers.data.ts` and export an array of cost center objects with id, name, and code properties ([link](https://github.com/fylein/fyle-mobile-app/pull/2123/files?diff=unified&w=0#diff-7de0329a6c26862e485c2274615a2f69250e4144bdf3abd27cc61d704384ec91R177-R205))
  - Create `src/app/core/mock-data/duplicate-sets.data.ts` and export an object of a duplicate set with id, created_at, updated_at, duplicate_expenses, and duplicate_of properties ([link](https://github.com/fylein/fyle-mobile-app/pull/2123/files?diff=unified&w=0#diff-08994ab67f55c1a7ea9ee90f7380a90fbd85abd94db9dbeee14f217f2cb1ebdfR1-R6))

## Clickup
https://app.clickup.com/t/85zt9ph0g

## Code Coverage
`10.17%` Statements 149/1465
`2.95%` Branches 46/1556
`6.36%` Functions 27/424
`10.16%` Lines 144/1416

## UI Preview
Please add screenshots for UI changes

### CODE REVIEW

File: src/app/core/mock-data/cost-centers.data.ts [hunk 1]
```diff
@@ -174,3 +174,32 @@ export const costCenterApiRes2: CostCenter[] = [
     updated_at: new Date('2022-05-19T11:58:29.364759+00:00'),
   },
 ];
+
+export const expectedCCdata = [
```
Comment (yksolanki9) [inline src/app/core/mock-data/cost-centers.data.ts:178]: type

File: src/app/core/mock-data/cost-centers.data.ts [hunk 2]
```diff
@@ -174,3 +174,32 @@ export const costCenterApiRes2: CostCenter[] = [
...
+  {
+    label: 'SMS1',
+    value: {
+      active: true,
+      code: null,
+      created_at: new Date('2019-06-24T15:12:04.002Z'),
```
Comment (yksolanki9) [inline src/app/core/mock-data/cost-centers.data.ts:184]: Type here should be string. Shouldnt be converted it to date

File: src/app/fyle/add-edit-expense/add-edit-expense.page.spec.ts [hunk 3]
```diff
@@ -1,26 +1,1020 @@
...
+      declarations: [AddEditExpensePage],
+      imports: [IonicModule.forRoot(), ReactiveFormsModule, FormsModule, RouterTestingModule, RouterModule],
+      providers: [
+        FormBuilder,
+        {
+          provide: ActivatedRoute,
+          useValue: {
+            snapshot: {
+              params: {
+                id: 'txyeiYbLDSOy',
```
Comment (yksolanki9) [inline src/app/fyle/add-edit-expense/add-edit-expense.page.spec.ts:259]: All the params will never be present together in the route. Try to replicate the behaviour of the app to cover all edge cases

File: src/app/fyle/add-edit-expense/add-edit-expense.page.spec.ts [hunk 4]
```diff
@@ -1,26 +1,1020 @@
...
+    ]);
+    fyCriticalPolicyViolationPopOverSpy.onWillDismiss.and.resolveTo({
+      data: {
+        action: 'primary',
+      },
+    });
+
+    modalController.create.and.resolveTo(fyCriticalPolicyViolationPopOverSpy);
+
+    const result = await component.continueWithCriticalPolicyViolation(criticalPolicyViolation2);
```
Comment (yksolanki9) [inline src/app/fyle/add-edit-expense/add-edit-expense.page.spec.ts:943]: Use fakeAsync and tick

**Review by yksolanki9 — APPROVED** [review]
Approving with minor comments

File: src/app/core/mock-data/cost-centers.data.ts (hunk 2 above)
Comment (jayfyle) [reply src/app/core/mock-data/cost-centers.data.ts:184]: The type is set as Date inside the model, will require changes in many places

File: src/app/fyle/add-edit-expense/add-edit-expense.page.spec.ts (hunk 4 above)
Comment (jayfyle) [reply src/app/fyle/add-edit-expense/add-edit-expense.page.spec.ts:943]: The presenting element returns expects a promise to be returned, using async await here is easier, plus the method does not resolve the data but returns it.

File: src/app/fyle/add-edit-expense/add-edit-expense.page.spec.ts [hunk 5]
```diff
@@ -1,26 +1,1011 @@
...
+      duplicate_detection_reason: [],
+      billable: [],
+      costCenter: [],
+      hotel_is_breakfast_provided: [],
+      project_dependent_fields: formBuilder.array([]),
+      cost_center_dependent_fields: formBuilder.array([]),
+    });
+
+    //@ts-ignore
+    component._isExpandedView = true;
```
Comment (Julias0) [inline src/app/fyle/add-edit-expense/add-edit-expense.page.spec.ts:511]: What is this?

File: src/app/fyle/add-edit-expense/add-edit-expense.page.spec.ts [hunk 6]
```diff
@@ -1,26 +1,1011 @@
...
+  }));
 
   it('should create', () => {
     expect(component).toBeTruthy();
   });
+
+  xit('scrollInputIntoView', () => {});
+
+  describe('goBack():', () => {
+    it('should back to the report automatically', () => {
```
Comment (Julias0) [inline src/app/fyle/add-edit-expense/add-edit-expense.page.spec.ts:524]: This test case is wrong.

bankTxn is not used - that part is dead code. Please remove that line.
persist_filters is used for a different scenario - the scenario for go back to report is when isRedirectedFromReport is `true`

File: src/app/fyle/add-edit-expense/add-edit-expense.page.spec.ts [hunk 7]
```diff
@@ -1,26 +1,1011 @@
...
+      activatedRoute.snapshot.params.persist_filters = true;
+      fixture.detectChanges();
+
+      navController.back.and.returnValue(null);
+
+      component.goBack();
+      expect(navController.back).toHaveBeenCalledTimes(1);
+    });
+
+    it('should back to txn if it is not redirected from report and filters are not persistent', () => {
```
Comment (Julias0) [inline src/app/fyle/add-edit-expense/add-edit-expense.page.spec.ts:535]: should back to txn -> should go back to my expenses page

File: capacitor.config.json [hunk 8]
```diff
@@ -0,0 +1 @@
+{}
```
Comment (Julias0) [inline capacitor.config.json:1]: what is this change? Please remove if not relevant

**Review by Julias0 — APPROVED** [review]

**Pull request merged**